File: test/codecarbon/code.java
class Main {
	public static void main(String[] args) {
		int repetitions = 1;
		if (args.length > 0) {
			try {
				repetitions = Integer.parseInt(args[0]);
			} catch (Exception e) {
				System.err.println("Argument non numérique, utilisation de 1 répétition.");
			}
		}

		System.out.println("Début du test Java - répétitions = " + repetitions);
		long t0 = System.currentTimeMillis();

		long totalPrimes = 0;
		for (int r = 0; r < repetitions; r++) {
			totalPrimes += countPrimes(2000000); // ajuster pour modifier la charge
		}

		long t1 = System.currentTimeMillis();
		System.out.println("Total primes (somme) : " + totalPrimes);
		System.out.println("Durée (ms) : " + (t1 - t0));
		System.out.println("Fin du test Java");
	}

	// compte le nombre de nombres premiers <= n (algorithme simple pour la charge CPU)
	static int countPrimes(int n) {
		int count = 0;
		for (int i = 2; i <= n; i++) {
			if (isPrime(i)) count++;
		}
		return count;
	}

	static boolean isPrime(int x) {
		if (x < 2) return false;
		if (x % 2 == 0) return (x == 2);
		int r = (int) Math.sqrt(x);
		for (int d = 3; d <= r; d += 2) {
			if (x % d == 0) return false;
		}
		return true;
	}
}

File: test/codecarbon/code.py
import time


def is_prime(n):
    if n < 2:
        return False
    if n % 2 == 0:
        return n == 2
    i = 3
    while i * i <= n:
        if n % i == 0:
            return False
        i += 2
    return True


def compute_primes(limit):
    primes = []
    for n in range(2, limit + 1):
        if is_prime(n):
            primes.append(n)
    return primes


def main():
    start = time.time()
    print("Démarrage du calcul des nombres premiers...")
    limit = 200000
    primes = compute_primes(limit)
    end = time.time()
    print(f"Nombre de premiers <= {limit} : {len(primes)}")
    print(f"Temps de calcul : {end - start:.3f} secondes")


if __name__ == "__main__":
    main()